src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -49,24 +114,47 @@ const serveFormReact: ControllerHandler = (_req, res) => {
       reactFrontendPath,
     },
   })
+  let reactHtml = readFileSync(path.join(reactFrontendPath, 'index.html'), {
+    encoding: 'utf8',
+  })
+  const formId = req.params.formId
+  const createMetatagsResult = await createMetatags({
+    formId,
+  })
+
+  // Failed to create metatags.
+  if (createMetatagsResult.isErr()) {
+    logger.error({
+      message: 'Error fetching metatags',
+      meta: {
+        action: 'handleRedirect',
+        ...createReqMeta(req),
+      },
+      error: createMetatagsResult.error,
+    })
+    reactHtml = reactHtml
+      // there are multiple __OG_TITLE__ and we are unable to use string.replaceAll so we use regexp
+      .replace(new RegExp('__OG_TITLE__', 'g'), 'FormSG')
+      .replace('__OG_DESCRIPTION__', '')
+  } else {
+    const { title, description } = createMetatagsResult.value
+    reactHtml = reactHtml
+      // there are multiple __OG_TITLE__ and we are unable to use string.replaceAll so we use regexp
+      .replace(new RegExp('__OG_TITLE__', 'g'), title)
+      .replace('__OG_DESCRIPTION__', description ?? '')
+      .replace('<title>FormSG</title>', `<title>${title}</title>`)
+      .replace('__OG_IMAGE__', 'og-img-metatag-publicform.png')
```

Comment:
It feels like the code to set the OG tags are duplicated, it might be nice to have a refactor with a pipelining approach, a given GET flow basically gets the og data, and eventually returns the file with replacement. For all get flows, the last step of the processing pipeline should be the same code 🤔